### PR TITLE
Classes 5 through 10

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: Class-5-(Functions).md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## In the prep work this week, the students learned:
+1. What a function is.
+1. What a parameter is.
+1. How to write a function with any number of parameters.
+1. How to call a function.
+1. What scope is and how it effects functions.
```
Comment (jimflores5) [inline Class-5-(Functions).md]: Typo: 'affects' instead of 'effects'.

File: Class-5-(Functions).md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+## In the prep work this week, the students learned:
+1. What a function is.
+1. What a parameter is.
```
Comment (jimflores5) [inline Class-5-(Functions).md:3]: Maybe mention arguments as well.

File: Class-5-(Functions).md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## Large Group Time (Instructor)
+### Reminders
+1. The grade for your first graded assignment will appear in Canvas.
+
+### Class 5 Topics That Require Careful Attention
+1. Conversation starters:
+   1. In a student's own words, can they explain what a function is?
+1. When is ``undefined`` returned?
+
```
Comment (jimflores5) [inline Class-5-(Functions).md:26]: How about adding something about scope? Examples via live coding, maybe.

Other topics:
a) Functions calling other functions.
b) Q & A for any issues with the chapter exercises.
c) Emphasize live coding. In particular, consider creating some function tasks to give to the students as "starters" or to use in walkthroughs.

File: Class-5-(Functions).md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+## Studio (TA Notes)
+
+### Class 5 Studio
```
Comment (jimflores5) [inline Class-5-(Functions).md:29]: Nice points!  I like the suggestion on guiding the students to use more efficient code.

Perhaps add another reminder that the students should access the studio via repl.it classroom rather than through the online book.

File: Class-6-(More-on-Functions).md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+## In the prep work this week, the students learned:
+1. What an anonymous function is and how to write one.
+2. What recursion is and how a function can call itself.
+
+## Announcements
+1. More info to come... when in doubt, check with the class Candidate Engagement Manager for announcements!
+1. The second graded assignment is open to you to start working on. Even if you can't solve everything right away, it is important to start working on it as you only have 2 weeks to do it!
```
Comment (jimflores5) [inline Class-6-(More-on-Functions).md]: Typo fix: "... is open for you to start..."
OR
If you want to remove the ending preposition, how about, "The second graded assignment is open, so you can start working on it after today's class."

Word swap suggestion: Replace "as you only" with "since you only".

File: Class-6-(More-on-Functions).md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+## Large Group Time (Instructor)
+
+### Class 6 Topics That Require Careful Attention
+1. Reinforce examples of when and why someone may use an anonymous function.
+1. Reinforce examples of when and why someone may choose the recursive solution.
+1. Conversation Starters:
```
Comment (jimflores5) [inline Class-6-(More-on-Functions).md:12]: The conversation starters should probably be at the top of the topics list.

File: Class-6-(More-on-Functions).md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+   1. What type of error occurs when attempting to use a value that is NOT a function as if it were one?
+   1. Explain the difference between named and anonymous functions, including an example of how an anonymous function can be used.
+
```
Comment (jimflores5) [inline Class-6-(More-on-Functions).md:18]: Other topics:
a) Q & A about the chapter exercises.
b) Encourage the instructor to use live coding examples when explaining anonymous functions.
c) While recursion is in the chapter, it should NOT receive a large amount of emphasis.  At this point in the students' learning journey, recursion pushes against the boundary between an engaging challenge and a discouraging frustration.

File: Class-7-(Objects-&-Math).md [hunk 8]
```diff
@@ -0,0 +1,21 @@
+## In the prep work this week, the students learned:
+1. That objects are collections of several key/value pairs.
+1. How to write object literals using bracket notation.
+1. How to access and modify the properties of an object.
+1. That JavaScript has built-in objects.
+1. How to use common built-in methods of the ``Math`` object.
+
+## Announcements
+1. More info to come... when in doubt, check with the class Candidate Engagement Manager!
+
```
Comment (jimflores5) [inline Class-7-(Objects-&-Math).md:11]: The second large assignment is due before the next class.

File: Class-7-(Objects-&-Math).md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+## Large Group Time (Instructor)
+
+### Class 7 Topics That Require Careful Attention
+1. What data type is an object? Students may need reminders and reinforcements that objects are built of other data types.
+1. Possible reasons to use an object.
+1. Why the ``Math`` object in particular? What can students do with the ``Math`` object?
+
```
Comment (jimflores5) [inline Class-7-(Objects-&-Math).md:21]: Other topics:
a) Show how the ``for...in`` loop iterates over the keys within an object.  This serves as a good intro for Graded Assignment #2.
b) Q & A for the exercises.
c) Intro for the studio.

File: Class-7-(Objects-&-Math).md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+## Announcements
+1. More info to come... when in doubt, check with the class Candidate Engagement Manager!
+
+## Large Group Time (Instructor)
+
+### Class 7 Topics That Require Careful Attention
+1. What data type is an object? Students may need reminders and reinforcements that objects are built of other data types.
+1. Possible reasons to use an object.
+1. Why the ``Math`` object in particular? What can students do with the ``Math`` object?
```
Comment (jimflores5) [inline Class-7-(Objects-&-Math).md]: Rephrasing suggestion: "What can students do with the ``Math`` object _besides_ simple rounding?"

Also, encourage examples for how ``Math`` allows us to accomplish not-actual-math actions like randomly selecting an array element.

File: Class-7-(Objects-&-Math).md [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+## Studio (TA Notes)
+
+### Class 7 Studio
```
Comment (jimflores5) [inline Class-7-(Objects-&-Math).md:24]: Remind TAs (again) to do some shoulder surfing/personal check-ins with each student. This far into the class, the TAs should be helping to guide their students away from inefficient code practices and other bad habits.

File: Class-8-(Modules).md [hunk 12]
```diff
@@ -0,0 +1,19 @@
+## In the prep work this week, the students learned:
+1. What a module is and why programmers use them.
+1. How to use npm and find a good module to use.
+
```
Comment (jimflores5) [inline Class-8-(Modules).md:5]: How to import and export modules.

File: Class-8-(Modules).md [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+## Announcements
+1. More info to come... when in doubt, check with the class Candidate Engagement Manager!
+1. Don't forget! Your second graded assignment is due soon!
```
Comment (jimflores5) [inline Class-8-(Modules).md]: Actually, the second assignment was due just before the Modules class started.

File: Class-8-(Modules).md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+1. What a module is and why programmers use them.
+1. How to use npm and find a good module to use.
+
+## Announcements
+1. More info to come... when in doubt, check with the class Candidate Engagement Manager!
+1. Don't forget! Your second graded assignment is due soon! 
+
+## Large Group Time (Instructor)
+
+### Class 8 Topics That Require Careful Attention
```
Comment (jimflores5) [inline Class-8-(Modules).md:11]: Other topics:
a) Live code the creation of a simple function (or two). Show how to export it and then import it into a different file.
b) How to structure the "path" when importing a module.
c) Q & A for the prep work exercises.
d) Non-coding studio intro.  Ask if anyone has felt like they are 'faking it'.  PEP TALK!

File: Class-8-(Modules).md [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+1. Reinforce possible use cases with modules.
+1. Double check that no one had any questions about npm. Remind students that npm is a valuable life skill for anyone pursuing a programming career.
+
+## Studio (TA Notes)
+
+### Class 8 Studio
```
Comment (jimflores5) [inline Class-8-(Modules).md:23]: Once created, add a link to a pdf copy of the TA notes for this studio.

File: Class-8-(Modules).md [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+1. Don't be afraid to offer your own experiences with imposter syndrome.
+1. Don't invalidate anyone's feelings.
```
Comment (jimflores5) [inline Class-8-(Modules).md:26]: Another possible item: Recognize that imposter syndrome is real and that the feelings it brings are normal.

File: Class-9-(Unit-Testing-&-TDD).md [hunk 17]
```diff
@@ -0,0 +1,17 @@
+## In the prep work this week, the students learned:
+1. What unit testing is and why it is important.
+1. How to use ``Jasmine`` for testing.
+
```
Comment (jimflores5) [inline Class-9-(Unit-Testing-&-TDD).md:5]: Also: What test driven development (TDD) is and how to use it to write solid code.

File: Class-9-(Unit-Testing-&-TDD).md [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+## Announcements
+1. More info to come... when in doubt, check with the class Candidate Engagement Manager!
+1. When your third graded assignment becomes available, make sure to check it out! Starting as soon as possible is key to finishing those graded assignments on time!
+
+## Large Group Time (Instructor)
+
+### Class 9 Topics That Require Careful Attention
+1. Reinforce why unit testing is important. If you can share an instance where properly testing your code saved you a huge headache, please do! 
+
```
Comment (jimflores5) [inline Class-9-(Unit-Testing-&-TDD).md:19]: Live coding, live coding, live coding. Experience is the best teacher, so the instructor should have some tasks prepared to demo unit testing as well as TDD.

Other topics:
a) Q & A for the chapter exercises or perform a walkthrough.
b) What "edge case" means, plus examples.
c) What does "describe" do compare to "it"? Review syntax, and encourage descriptive feedback messages.
d) Studio intro: The exercises practiced using unit tests to validate existing code. The studio flips this to a TDD scenario - the students must write tests BEFORE creating any of the code.

File: Class-9-(Unit-Testing-&-TDD).md [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+## Studio (TA Notes)
+
+### Class 9 Studio
```
Comment (jimflores5) [inline Class-9-(Unit-Testing-&-TDD).md:22]: Other items:
a) Reinforce the benefits of writing tests before the code.
b) If many of the students have the same questions or problems with writing their first test, consider going through that as a group.
c) This would be a good studio for the students to work on in pairs. Each student still needs to submit their own final project, but collaborating on the code helps with the learning curve.

File: Class-10-(Scope-Exceptions-Types).md [hunk 20]
```diff
@@ -0,0 +1,23 @@
+## In the prep work this week, the students learned:
+1. More about scope. Scope was covered briefly in the Functions chapter, however, now students have covered it more in-depth in terms of the order code executes in.
```
Comment (jimflores5) [inline Class-10-(Scope-Exceptions-Types).md]: Rephrasing suggestion: "Scope was covered briefly in the Functions chapter, but the students have now covered it in more depth."

File: Class-10-(Scope-Exceptions-Types).md [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+1. What exceptions are.
+1. How to raise an exception.
+1. More about what a primitive data type is.
+
+## Announcements
+1. More info to come... when in doubt, check with your class Candidate Engagement Manager!
+
```
Comment (jimflores5) [inline Class-10-(Scope-Exceptions-Types).md:10]: The third graded assignment opens after this class. Feel free to start working on the smaller parts.

File: Class-10-(Scope-Exceptions-Types).md [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+## Large Group Time (Instructor)
+
+### Class 10 Topics That Require Careful Attention
+1. Conversation starters:
+   1. What is the difference between different types of errors.
+   1. What are some more common errors?
+   1. How do we raise an exception?
+   1. How do we handle exceptions?
+
```
Comment (jimflores5) [inline Class-10-(Scope-Exceptions-Types).md:24]: Other topics:
a) What does "variable hoisting" mean, and what does it look like?
b) Why would we want to throw an exception?
c) Why should we try to avoid variable shadowing?
d) Q & A for the exercises.
e) Studio intro: Non-coding activity this week on Strategic Debugging.

File: Class-10-(Scope-Exceptions-Types).md [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+## Studio (TA Notes)
+
+### Class 10 Studio
```
Comment (jimflores5) [inline Class-10-(Scope-Exceptions-Types).md:27]: Once created, add a link to a pdf copy of the TA notes for this activity.

File: Class-10-(Scope-Exceptions-Types).md [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+1. Don't be afraid to share your own experience!
+1. This studio is aimed on helping people develop processes for more strategically debugging their code. If after this studio, they are still struggling with that, don't hesitate to gently remind them of the topics covered here.
```
Comment (jimflores5) [inline Class-10-(Scope-Exceptions-Types).md]: Rephrasing suggestion for the second sentence: "After this studio, if they are still struggling, do not hesitate to gently remind them of the topics covered here."
OR 
"If they still struggle with debugging after this studio, gently remind them of the topics covered here."

File: Class-5-(Functions).md (hunk 1 above)
Comment (gildedgardenia) [reply Class-5-(Functions).md]: Thanks!

File: Class-5-(Functions).md (hunk 2 above)
Comment (gildedgardenia) [reply Class-5-(Functions).md:3]: Done! Thanks!

File: Class-6-(More-on-Functions).md (hunk 5 above)
Comment (gildedgardenia) [reply Class-6-(More-on-Functions).md]: Done! Thanks!

File: Class-6-(More-on-Functions).md (hunk 6 above)
Comment (gildedgardenia) [reply Class-6-(More-on-Functions).md:12]: Done! Thanks!

File: Class-7-(Objects-&-Math).md (hunk 8 above)
Comment (gildedgardenia) [reply Class-7-(Objects-&-Math).md:11]: Done! Thanks!

File: Class-8-(Modules).md (hunk 13 above)
Comment (gildedgardenia) [reply Class-8-(Modules).md]: Thanks!

File: Class-10-(Scope-Exceptions-Types).md (hunk 21 above)
Comment (gildedgardenia) [reply Class-10-(Scope-Exceptions-Types).md:10]: Done! Thanks!

File: Class-10-(Scope-Exceptions-Types).md (hunk 24 above)
Comment (gildedgardenia) [reply Class-10-(Scope-Exceptions-Types).md]: Done! Thanks!

File: Class-10-(Scope-Exceptions-Types).md (hunk 20 above)
Comment (gildedgardenia) [reply Class-10-(Scope-Exceptions-Types).md]: Done! Thanks!

File: Class-5-(Functions).md (hunk 3 above)
Comment (gildedgardenia) [reply Class-5-(Functions).md:26]: Done! Thanks!

File: Class-5-(Functions).md (hunk 4 above)
Comment (gildedgardenia) [reply Class-5-(Functions).md:29]: Done! Thanks!

File: Class-6-(More-on-Functions).md (hunk 7 above)
Comment (gildedgardenia) [reply Class-6-(More-on-Functions).md:18]: Done! Thanks!

File: Class-7-(Objects-&-Math).md (hunk 9 above)
Comment (gildedgardenia) [reply Class-7-(Objects-&-Math).md:21]: Done! Thanks!

File: Class-7-(Objects-&-Math).md (hunk 10 above)
Comment (gildedgardenia) [reply Class-7-(Objects-&-Math).md]: Done! Thanks!

File: Class-7-(Objects-&-Math).md (hunk 11 above)
Comment (gildedgardenia) [reply Class-7-(Objects-&-Math).md:24]: Done! Thanks!

File: Class-8-(Modules).md (hunk 12 above)
Comment (gildedgardenia) [reply Class-8-(Modules).md:5]: Done! Thanks!

File: Class-8-(Modules).md (hunk 14 above)
Comment (gildedgardenia) [reply Class-8-(Modules).md:11]: Done! Thanks!

File: Class-8-(Modules).md (hunk 16 above)
Comment (gildedgardenia) [reply Class-8-(Modules).md:26]: Done! Thanks!

File: Class-8-(Modules).md (hunk 15 above)
Comment (gildedgardenia) [reply Class-8-(Modules).md:23]: What way have people been doing this so far? Google Drive link?

File: Class-9-(Unit-Testing-&-TDD).md (hunk 17 above)
Comment (gildedgardenia) [reply Class-9-(Unit-Testing-&-TDD).md:5]: Done! Thanks!

File: Class-9-(Unit-Testing-&-TDD).md (hunk 18 above)
Comment (gildedgardenia) [reply Class-9-(Unit-Testing-&-TDD).md:19]: Done! Thanks!

File: Class-9-(Unit-Testing-&-TDD).md (hunk 19 above)
Comment (gildedgardenia) [reply Class-9-(Unit-Testing-&-TDD).md:22]: Done! Thanks!

File: Class-10-(Scope-Exceptions-Types).md (hunk 22 above)
Comment (gildedgardenia) [reply Class-10-(Scope-Exceptions-Types).md:24]: Done! Thanks!

File: Class-10-(Scope-Exceptions-Types).md (hunk 23 above)
Comment (gildedgardenia) [reply Class-10-(Scope-Exceptions-Types).md:27]: See above question. What do you think is the best way to do this?

File: Class-8-(Modules).md (hunk 15 above)
Comment (jimflores5) [reply Class-8-(Modules).md:23]: Blake figured out a way to add a pdf file to the intro-to-prof-web-dev repo and then link to it from the wiki pages.  There is a folder in the master branch called "wiki-materials", and he added screenshots and the TA notes there.  

This was done before we established a separate repo for wiki content, so I am not sure how we would add files now.
